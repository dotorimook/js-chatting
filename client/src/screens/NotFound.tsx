import React, { FC } from 'react';
import IComponentProps from 'interfaces/IComponentProps';
import { FrameCenter } from 'components/Frame';
import image from 'assets/img/notFound/0.svg';

const NotFound:FC<IComponentProps> = (props:IComponentProps) => {
  return (
    <FrameCenter className='msg-frame'>
      <img className='title-img' src={image} alt='logo'/>
      <div className='msg'>
        잘못된 페이지입니다.
      </div>
    </FrameCenter>
  )
}
export default NotFound;