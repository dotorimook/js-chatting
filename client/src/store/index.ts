import Error from "./models/Error";
import Loading from "./models/Loading";
import LoginInfo from "./models/LoginInfo";
import RoomList from "./models/screens/RoomList";
import ChatRoom from "./models/screens/ChatRoom";
import Login from "./models/screens/Login";

const store = {
  loginInfo: new LoginInfo(),
  loading: new Loading(),
  error: new Error({}),

  /** screens */
  screenLogin: new Login(),
  screenRoomList: new RoomList(),
  screenChatRoom: new ChatRoom(),
};

export default store;