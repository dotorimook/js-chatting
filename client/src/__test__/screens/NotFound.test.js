import React from 'react';
import 'setupTest';
import {shallow} from 'enzyme';
import NotFound from 'screens/NotFound';
import { Provider } from 'mobx-react';
import store from 'store';

describe('NotFound Test', ()=>{
  let ui = shallow(
    <Provider {...store}>
      <NotFound/>
    </Provider>
  );

  it('matches snapshot', () => {
    expect(ui).toMatchSnapshot();
  })
});