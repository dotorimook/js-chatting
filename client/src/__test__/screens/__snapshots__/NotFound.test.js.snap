// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Test matches snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "error": Error {
        "code": "UNDEFINED",
        "msg": "알 수 없는 에러",
        "visible": false,
      },
      "loading": Loading {
        "isLoading": false,
      },
      "loginInfo": LoginInfo {
        "isFirstTry": true,
      },
      "screenChatRoom": ChatRoom {
        "socket": null,
      },
      "screenLogin": Login {},
      "screenRoomList": RoomList {},
    }
  }
>
  <NotFound />
</ContextProvider>
`;
